src/cml.js

```diff
@@ -72,12 +68,18 @@ const getDriver = (opts) => {
 
 class CML {
   constructor(opts = {}) {
+    let { repo } = opts;
     const { driver, repo, token } = opts;
 
-    this.repo = uriNoTrailingSlash(repo || gitRemoteUrl()).replace(
-      /\.git$/,
-      ''
-    );
+    try {
+      if (!repo) repo = gitRemoteUrl();
+    } catch (err) {
+      err.message +=
+        ', you might need to manually specify the --repo option or install Git';
```

Comment:
super minor: maybe newline instead of `, `

Reply:
@DavidGOrtega asked me to let him solve #618 by replacing the `git` command-line tool by a JavaScript library, so I have closed this pull request. Nevertheless, your comment is still on topic and useful. Can you please move it to #606?

Reply:
Historical note: allegedly, this pull request was closed in favor of a then nonexistent https://github.com/iterative/cml/pull/1003.